EcoSoapBank/Custom Views/KeyboardHandlingViewController.swift

```diff
@@ -0,0 +1,106 @@
+
+    private func configure() {
+        contentView.translatesAutoresizingMaskIntoConstraints = false
+        scrollView.addSubview(contentView)
+
+        scrollView.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(scrollView)
+
+        NSLayoutConstraint.activate([
+            contentView.topAnchor.constraint(equalTo: scrollView.contentLayoutGuide.topAnchor),
+            contentView.leadingAnchor.constraint(equalTo: scrollView.contentLayoutGuide.leadingAnchor),
+            contentView.trailingAnchor.constraint(equalTo: scrollView.contentLayoutGuide.trailingAnchor),
+            contentView.bottomAnchor.constraint(equalTo: scrollView.contentLayoutGuide.bottomAnchor),
+
+            contentView.widthAnchor.constraint(equalTo: scrollView.frameLayoutGuide.widthAnchor),
+
+            scrollView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+            scrollView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
+            scrollView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
+            scrollView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
+        ])
+
+        let weakHeightConstraint = contentView.heightAnchor.constraint(equalTo: scrollView.frameLayoutGuide.heightAnchor)
+        weakHeightConstraint.priority = .defaultLow
+        weakHeightConstraint.isActive = true
+
+        scrollView.contentInsetAdjustmentBehavior = .never
+        scrollView.keyboardDismissMode = .interactive
+
+        setupKeyboardNotifications()
+    }
+
+    // MARK: - Keyboard Handling
+
+    private func setupKeyboardNotifications() {
+        NotificationCenter.default
+            .publisher(for: UIResponder.keyboardWillShowNotification)
+            .merge(with: NotificationCenter.default
+                .publisher(for: UIResponder.keyboardWillHideNotification))
+            .sink(receiveValue: moveForKeyboard(notification:))
+            .store(in: &cancellables)
+    }
+
+    private func moveForKeyboard(notification: Notification) {
+        let key = UIResponder.keyboardFrameEndUserInfoKey
+        if let info = notification.userInfo,
+            let keyboardFrame = info[key] as? CGRect {
+            scrollView.contentInset = UIEdgeInsets(
+                top: 0,
+                left: 0,
+                bottom: keyboardFrame.height,
+                right: 0)
+        }
+    }
+}
+
+extension KeyboardHandlingViewController: UITextViewDelegate, UITextFieldDelegate {
+    func textViewDidBeginEditing(_ textView: UITextView) {
+        DispatchQueue.main.async { [weak scrollView] in
+            scrollView?.scrollRectToVisible(textView.frame, animated: true)
+        }
+    }
+
+    func textFieldDidBeginEditing(_ textField: UITextField) {
+        DispatchQueue.main.async { [weak scrollView] in
+            scrollView?.scrollRectToVisible(textField.frame, animated: true)
+        }
+    }
+}
```

Comment:
This seems way more elegant than what I had. I guess I was over-complicating things.